test/quantization/fx/test_model_report_fx.py

```diff
@@ -1160,3 +1168,59 @@ def forward(self, x):
+        # then create model report instance with detector
+        with override_quantized_engine('fbgemm'):
+
+            test_input_weight_detector = InputWeightEqualizationDetector(0.4)
+            detector_set = set([test_input_weight_detector])
+            # prepare the model for callibration
+            prepared_for_callibrate_model, model_report = self._get_prepped_for_calibration_model(detector_set)
+
+            # now we actually callibrate the model
+            example_input = torch.arange(27).reshape((1, 3, 3, 3))
+            example_input = example_input.to(torch.float)
+
+            prepared_for_callibrate_model(example_input)
+
+            # now get the report by running it through ModelReport instance
+            generated_report = model_report.generate_model_report(prepared_for_callibrate_model, True)
+
+            # check that sizes are appropriate only 1 detector
+            self.assertEqual(len(generated_report), 1)
+
+            # get the specific report for input weight equalization
+            input_weight_str, input_weight_dict = generated_report[test_input_weight_detector.get_detector_name()]
+
+            # we should have 4 layers looked at since 4 conv / linear layers
+            self.assertEqual(len(input_weight_dict), 4)
+
+            # we can validate that the max and min values of the detector were recorded properly for the first one
+            # this is because no data has been processed yet, so it should be values from original input
+
+            example_input = example_input.reshape((3, 3, 3))  # reshape input
+            for module_fqn in input_weight_dict:
+                # look for the first linear
+                if "block1.linear" in module_fqn:
+                    block_1_lin_recs = input_weight_dict[module_fqn]
+                    # get input range info and the channel axis
+                    input_range_info = block_1_lin_recs[InputWeightEqualizationDetector.INPUT_INFO_KEY]
+                    ch_axis = block_1_lin_recs[InputWeightEqualizationDetector.CHANNEL_KEY]
+
+                    # ensure that the min and max values extracted match properly
+                    example_min, example_max = torch.aminmax(example_input, dim=ch_axis)
+                    dimension_min = torch.amin(example_min, dim=ch_axis)
+                    dimension_max = torch.amax(example_max, dim=ch_axis)
+
+                    self.assertEqual(input_range_info[InputWeightEqualizationDetector.PER_CHANNEL_MIN_KEY], dimension_min)
+                    self.assertEqual(input_range_info[InputWeightEqualizationDetector.PER_CHANNEL_MAX_KEY], dimension_max)
+
+                    # make sure the global min and max were correctly recorded and presented
+                    self.assertEqual(input_range_info[InputWeightEqualizationDetector.GLOBAL_MIN_KEY], min(dimension_min))
+                    self.assertEqual(input_range_info[InputWeightEqualizationDetector.GLOBAL_MAX_KEY], max(dimension_max))
```

Comment:
I feel there are a lot more things we can check here. E.g. Can we add a test for the comparison stat? We can assert whether it is in fact equal to the sqrt ratio given the input and weight ranges